lib/src/highlight/node.dart

```diff
@@ -51,4 +51,37 @@ extension MyNode on Node {
 
     return null;
   }
+
+  Node splitNewLines() {
+    children = children?.splitNewLines();
```

Comment:
Are they mutable? 🤦

This method is misleading because it splits children but not this node. Also a node's text is currently split in the extension on `List<Node>` which is not right, it should be one in an extension on a node itself. So let's do this:

1. Make it split both this node and children. For this, return `List<Node>`.
2. If `value == null`, return a list of a single new `Node` with split `children`.
3. Otherwise return a list of new nodes split with newlines.

Neither way do we mutate a node.